This file is part of MXE.
See index.html for further information.

Some functions in netioapi.h were only added in Windows Vista. This patch removes
function definitions for if_nametoindex and if_indextoname on older API
versions, so they don't conflict with replacement implementations in libraries
(e.g. glib).

This is fixed on the 4.x branch of mingw-w64.

diff -Naur mingw-w64-v3.3.0.orig/mingw-w64-headers/include/netioapi.h mingw-w64-v3.3.0/mingw-w64-headers/include/netioapi.h
--- mingw-w64-v3.3.0.orig/mingw-w64-headers/include/netioapi.h	2015-01-17 00:07:51.030935703 +0100
+++ mingw-w64-v3.3.0/mingw-w64-headers/include/netioapi.h	2015-01-17 01:09:47.546591536 +0100
@@ -311,6 +311,7 @@
   PNET_LUID InterfaceLuid
 );
 
+#if (_WIN32_WINNT >= 0x0600)
 PCHAR WINAPI if_indextoname(
   NET_IFINDEX InterfaceIndex,
   PCHAR InterfaceName
@@ -319,6 +320,7 @@
 NET_IFINDEX WINAPI if_nametoindex(
   PCSTR InterfaceName
 );
+#endif
 
 NETIO_STATUS WINAPI ConvertInterfaceGuidToLuid(
   const GUID *InterfaceGuid,
